src/Features/Core/Portable/AddDebuggerDisplay/AbstractAddDebuggerDisplayCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.AddImports;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.AddDebuggerDisplay
+{
+    internal abstract class AbstractAddDebuggerDisplayCodeRefactoringProvider<TTypeDeclarationSyntax, TMethodDeclarationSyntax> : CodeRefactoringProvider
+        where TTypeDeclarationSyntax : SyntaxNode
+        where TMethodDeclarationSyntax : SyntaxNode
+    {
+        private static readonly SyntaxAnnotation s_trackingAnnotation = new SyntaxAnnotation();
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var type = await context.TryGetRelevantNodeAsync<TTypeDeclarationSyntax>().ConfigureAwait(false);
```

Comment:
nice!  tagging @petrroll  :)   great that we can use this here.

Reply:
I'm not sure if the helper handles headers of Type declarations correctly. Will double check.